cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/WorkflowHttpHandler.java

```diff
@@ -142,7 +143,8 @@ public void resumeWorkflowRun(HttpRequest request, HttpResponder responder,
    */
   private ProgramController getProgramController(String namespaceId, String appId,
                                                  String workflowName, String runId) throws NotFoundException {
-    ProgramId id = new ProgramId(namespaceId, appId, ProgramType.WORKFLOW, workflowName);
+    ApplicationId applicationId = new ApplicationId(namespaceId, appId, getLatestAppVersion(namespaceId, appId));
```

Comment:
I feel that, these APIs need special handling. They're passed in runID, and with LCM, the latest version might not match the old runs. As I discussed with @albertshau last week, we need to fetch runID ignoring version, then back-words validate the version and stuff